### PR TITLE
add watch_topic

### PR DESCRIPTION
**Pull request by glanzel**
### Summary of changes

adds api call for: user gets notifications about a topic.

## Checklist

- [x] Changes represent a *discrete update*
- [x] Commit messages are meaningful and descriptive
- [x] Changeset does not include any extraneous changes unrelated to the discrete change

### CODE REVIEW

**Comment by glanzel** [comment]
sorry for the one useless commit. I just didn't excpected the pull request gets resolved right after i tried out some stuff . Now everything is back in place.

**Comment by glanzel** [comment]
I just wanted to clarify that my commit was ready for pulling  since 2020/05/14. 
I think it is a usefull commit as it adds additional features. I needed this change for my work and would like to see it in your respository. As I am new to use Pull request in github please feel free to give me hints i something is wrong with my request.

**Comment by bennylope** [comment]
The feature would be a nice addition, but there's some unnecessary commits and changes here.

I started to clean that up - which is not a big deal - but I'm confused about why the `api_user` is reset in the `watch_topic` method.

File: pydiscourse/client.py [hunk 1]
```diff
@@ -198,14 +198,14 @@ def trust_level(self, userid, level):
...
-    def suspend(self, userid, duration, reason):
+    def suspend(self, userid, duration, reason): 
         """
         Suspend a user's account
 
         Args:
             userid: the Discourse user ID
             duration: the length of time in days for which a user's account
-                    should be suspended
+                    should be suspended (365000 for forever)
```
Comment (bennylope) [inline pydiscourse/client.py:208]: ?

File: pydiscourse/client.py [hunk 2]
```diff
@@ -551,6 +551,19 @@ def delete_topic(self, topic_id, **kwargs):
...
+        """
+
+        Args:
+            topic_id:
+            **kwargs:
+
+        Returns:
+
+        """
+        self.api_username = username
```
Comment (bennylope) [inline pydiscourse/client.py:564]: Why is this resetting the `api_username`? That's the username to connect to the API with, and it's already set when creating the client instance. Changing it here would presumably result in authentication errors trying to make requests to the Discourse API...

**Comment by bennylope** [comment]
Closing this, if requested changes are made will re-open.

**Comment by glanzel** [comment]
yes sure,

i am very sorry that my communication was so bad.
I fact the resetting the api_username is need for the functionality and
works if the api_user is system
(dont know why) but dont works if the api_user is some else.
So it's probably nothing for the official pydiscourse like that.
Unfortunately i could figure out how to do this in a better way with the
given discourse api
so i lost my motivation.

Sorry for the additional work for you.

Greeting grischan





Am Mi., 11. Nov. 2020 um 00:37 Uhr schrieb Ben Lopatin <
notifications@github.com>:

**Pull request closed without merging**